### PR TITLE
feat: added semantic releases

### PR DESCRIPTION
**Pull request by MariusSchmid**
I added following feature regarding: https://github.com/LordFckHelmchen/wrath-and-glory-space-hulk-generator/issues/14

* pushes to main will release
* increment version number according to the conventional commits
* create CHANGELOG.md automatically

### CODE REVIEW

File: .github/workflows/code_checks.yml [hunk 1]
```diff
@@ -62,3 +62,24 @@ jobs:
...
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Python Semantic Release
+        uses: python-semantic-release/python-semantic-release@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (LordFckHelmchen) [inline .github/workflows/code_checks.yml:85]: Add a newline:
```suggestion
          github_token: ${{ secrets.GITHUB_TOKEN }}

```

File: pyproject.toml [hunk 2]
```diff
@@ -80,3 +80,6 @@ convention = "numpy"  # Accepts: "google", "numpy", or "pep257".
 [build-system]
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
+
+[tool.semantic_release]
+version_variables = ["pyproject.toml:version"]
```
Comment (LordFckHelmchen) [inline pyproject.toml:85]: Add missing newline at the end of the file.
```suggestion
version_variables = ["pyproject.toml:version"]

```

**Review by LordFckHelmchen — APPROVED** [review]
Hey Mariüs! Thanks a lot for the contribution! My first contribution from someone else!

**Comment by MariusSchmid** [comment]
@LordFckHelmchen Do you have an idea why the unittests and ruff linting is not working?

**Comment by LordFckHelmchen** [comment]
> @LordFckHelmchen Do you have an idea why the unittests and ruff linting is not working? Can you please have a look? It seems its not related to my changes.

Not related. Tried to turn it off and on again. Worked.

**Comment by MariusSchmid** [comment]
wohooo... @LordFckHelmchen can you please merge? I do not have write access 
grafik

**Pull request merged**